### PR TITLE
Store and pop receiver variable bytecode

### PR DESCRIPTION
**Pull request by PalumboN**
Adding supports for bytecodes `200 201 202 203 204 205 206 207`

## Merge PR in the VM before

https://github.com/pharo-project/pharo-vm/pull/711

### CODE REVIEW

File: Druid-Tests/DRProductionBytecodeTest.class.st [hunk 1]
```diff
@@ -1436,3 +1435,141 @@ DRProductionBytecodeTest >> testReturnTopFromMethod [
...
+
+			"Execute the druid's compiled code"
+			generator value.
+
+			"Then return without druid's compiled code"
+			cogit genUpArrowReturn ].
+
+	object := self newObjectWithSlots: index + 1.
+
+	machineSimulator temporaryRegisterValue: memory trueObject.
```
Comment (guillep) [inline Druid-Tests/DRProductionBytecodeTest.class.st:1472]: This goes to the remembered set in our case right?

File: Druid/DRBytecodeIRGenerator.class.st [hunk 2]
```diff
@@ -303,6 +305,27 @@ DRBytecodeIRGenerator >> interpretPushWith: aRBMessageNode [
...
+		   addInstructionWithNoResultFrom: aRBMessageNode
+		   instructionKind: DRCall
+		   operands: {
+				   'ceStoreCheckTrampoline'.
+				   obj }) beMapped saveLinkReg
+
+	"	
+	cogit 
+		evaluateTrampolineCallBlock: [cogit CallRT: ceStoreCheckTrampoline]
+		protectLinkRegIfNot: inFrame.
```
Comment (guillep) [inline Druid/DRBytecodeIRGenerator.class.st:325]: Ok, we should review this later, I wonder why we need a special trampoline call.

File: Druid/DRIRGenerator.class.st [hunk 3]
```diff
@@ -311,6 +311,8 @@ DRIRGenerator >> initializeSpecialCases [
 	
 	
 	specialCases at: #longAt:put: put: #interpretMachineWordAtPutWith:.
+	specialCases at: #unsignedLong64At:put: put: #interpretMachineWordAtPutWith:.
+	specialCases at: #unsignedLongAt:put: put: #interpretMachineWordAtPutWith:.
```
Comment (guillep) [inline Druid/DRIRGenerator.class.st:315]: Ah we should do a pass on all the loads and stores after all the branches are integrated :)

File: Druid/DRIRGenerator.class.st [hunk 4]
```diff
@@ -1806,6 +1808,8 @@ DRIRGenerator >> visitMessageNode: aRBMessageNode receiver: receiver [
 { #category : #visiting }
 DRIRGenerator >> visitMethodNode: aRBMethodNode [
 
+	(aRBMethodNode hasPragmaNamed: #doNotGenerate) ifTrue: [ 1 halt ].
```
Comment (guillep) [inline Druid/DRIRGenerator.class.st:1811]: Interesting, have a case?
Why not the same pragma that in the other PR for the halt?

**Review by guillep — APPROVED** [review]
Good!

File: Druid-Tests/DRProductionBytecodeTest.class.st (hunk 1 above)
Comment (PalumboN) [reply Druid-Tests/DRProductionBytecodeTest.class.st:1472]: Nop, the object is in the new space, nothing to remember.

File: Druid/DRBytecodeIRGenerator.class.st (hunk 2 above)
Comment (PalumboN) [reply Druid/DRBytecodeIRGenerator.class.st:325]: Now we are generating something like that, I forget the comment.

File: Druid/DRIRGenerator.class.st (hunk 4 above)
Comment (PalumboN) [reply Druid/DRIRGenerator.class.st:1811]: Hahahah I used it to know quickly where the meta-compilation scapes from the "real" interpreter code.

Good to find missing special cases. We can remove it (or not... should never happen I think 🤔 )

**Pull request merged**